pandas/core/arrays/masked.py

```diff
@@ -112,7 +113,7 @@ class BaseMaskedArray(OpsMixin, ExtensionArray):
     _internal_fill_value: Scalar
     # our underlying data and mask are each ndarrays
     _data: np.ndarray
-    _mask: npt.NDArray[np.bool_]
+    _mask: BitMaskArray
```

Comment:
Maybe we can make this a property and override the getter?

Reply:
What are you thinking we override the getter with? Are you thinking of special logic or still to just treat it like a normal attribute?

Reply:
I was thinking that you can make ``_mask`` a property with a getter that calls ``to_numpy`` on a internal ``_bitmask`` attribute where we'll store the actual mask.

Reply:
Ah I see. The problem is sometimes we want to access the actual bitmask object (I.e. to forward to another constructor for reference counting). The to_numpy calls are only required when we are implicitly using numpy functionality in the code (granted this is pretty often)

Reply:
In theory it would be great to eliminate `to_numpy()` calls. With `__or__` being a good example, it would be preferable to do that bitwise between two buffers if they are bitwise masks with the same length.

In practice I was having trouble properly exposing that through the Cython classes so am hoping to leave as a future optimization

Reply:
We might need to keep `_mask` as a property that returns the numpy byte mask (this is somewhat semi-public), as Thomas mentioned, and add another attribute for the new bitmask (so essentially where you now changed `._mask` to `._mask.to_numpy()`, it would stay `_mask`, and anywhere you now uses just `._mask`, that would become something like `._bitmask`)

See https://github.com/pandas-dev/pandas/issues/34873 for some context on this

Reply:
Good to know - I'm going to hold off on this until the end, maybe a follow up depending on how much of a change it requires. It is definitely feasible to do as you've suggested just want to balance that against a huge initial diff

Reply:
I'm a bit conflicted on this. The big downside to keeping `._mask` call `to_numpy()` is that that is a really expensive conversion, so it can cause performance regression unless you really root it out. Assuming it is not a ton of downstream libraries that have assumed `._mask` is an ndarray maybe we can coordinate with them to update to `._mask.to_numpy()` and maybe even `._mask.as_bitmask()`

Reply:
I think what we could do for now is ask any downstream libraries that have been using this to switch to `isna()`, which has and will continue to return a NumPy array with this PR